src/main/java/uk/gov/register/presentation/resource/EntryResource.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.register.presentation.resource;
+
+import io.dropwizard.jersey.caching.CacheControl;
+import org.postgresql.util.PSQLException;
+import uk.gov.register.presentation.DbEntry;
+import uk.gov.register.presentation.dao.Entry;
+import uk.gov.register.presentation.dao.EntryDAO;
+import uk.gov.register.presentation.dao.RecentEntryIndexQueryDAO;
+import uk.gov.register.presentation.representations.ExtraMediaType;
+import uk.gov.register.presentation.view.AttributionView;
+import uk.gov.register.presentation.view.SingleEntryView;
+import uk.gov.register.presentation.view.ViewFactory;
+
+import javax.inject.Inject;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import java.util.Optional;
+import java.util.function.Function;
+
+@Path("/entry")
+public class EntryResource {
+
+    private final EntryDAO entryDAO;
+    private final ViewFactory viewFactory;
+    private final RecentEntryIndexQueryDAO queryDAO;
+    private final RequestContext requestContext;
+
+
+    @Inject
+    public EntryResource(EntryDAO entryDAO, ViewFactory viewFactory, RecentEntryIndexQueryDAO queryDAO, RequestContext requestContext) {
+        this.entryDAO = entryDAO;
+        this.viewFactory = viewFactory;
+        this.queryDAO = queryDAO;
+        this.requestContext = requestContext;
+    }
+
+    @GET
+    @Path("/{entry-number}")
+    @Produces({ExtraMediaType.TEXT_HTML, MediaType.APPLICATION_JSON})
+    @CacheControl(immutable = true)
+    public AttributionView findByEntryNumber(@PathParam("entry-number") int entryNumber) {
+        try {
+            Optional<Entry> entry = entryDAO.findByEntryNumber(entryNumber);
+            return entry.map(viewFactory::getNewEntryView).orElseThrow(NotFoundException::new);
+        } catch (Throwable e) {
+            if (e.getCause() instanceof PSQLException && ((PSQLException) e.getCause()).getSQLState().equals("42P01")) {
```

Comment:
`42P01` is postgres error code which is returned in case of queried table does not exist. There is a unit test case to explain this, we can also extract it as a constant. 
Special case is required so that we don't have deployment dependencies between indexer and presentation. If new schema is not available the old resource will respond otherwise the new resource response will be returned.